CMakeLists.txt

```diff
@@ -70,14 +70,43 @@ if (TI_WITH_CUDA)
     set(CUDA_ARCH "cuda")
 endif()
 
-find_program(CLANG_EXECUTABLE NAMES clang clang-7 clang-8 clang-9 clang-10)
+if (CLANG_PATH)
```

Comment:
I'm a bit confused of what's the best approach for setting clang paths now.. AFAICT, we have env var `CC`, `CXX`, and cmake variables `CLANG_PATH` and/or `CMAKE_C_COMPILER`?

Reply:
Yeah, it's indeed confusing... The point is that we need two `clang` to compile taichi itself and the LLVM runtime respectively. 

Specifically, `CMAKE_C_COMPILER` determines the compiler for taichi, and if not set it is obtained from ENV `CC`. Same for `CMAKE_CXX_COMPILER` and `CXX`. They have no effect on the compiler of the LLVM runtime because that is detected from `find_program` as a normal binary. So the new `CLANG_EXCUTABLE` is used to affect the behavior of `find_program`.

Actually I prefer to use the same `clang` for both if possible... But it will break our CI due to the strange search order of `clang` on macOS.

Reply:
@sjwsl  is it possible that we change the search order so that we can use the same clang? ;) 
We indeed want to use only one clang IMHO.

Reply:
Agreed! So can we completely prohibit the GNU compilers? Otherwise we still need to keep these confusing parameters.

Reply:
@sjwsl Sure I think we made it clear in the doc that we only support clang so it should be safe to prohibit gcc as long as we throw proper error messages?

Reply:
Yep. Anyway `clang` is needed for llvm runtime, so it will not bring more troubles. :)